### PR TITLE
Add the boring typescript configs

### LINKED ISSUE #15: Add basic CLI scaffolding
To have mocked CLI commands and a basic typescript setup

### PR DESCRIPTION
**Pull request by konnov**
Closes #15. This PR adds plenty of boring config files, basically a "hello, world" of typescript + github. Even though this PR has 5.2 KLOC, there is not much happening, if you look at the change set. A massive change is caused by the license file autogenerated by `source-licenser`.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+jobs:
+  unit-tests:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      fail-fast: false
+      matrix:
+        operating-system: [ubuntu-latest]
+        # macos is burning my free minutes way too fast
+        #operating-system: [ubuntu-latest, macos-latest]
```
Comment (Kukovec) [inline .github/workflows/main.yml:18]:
```suggestion
```

Comment (konnov) [reply .github/workflows/main.yml:18]: What is the change here?

**Comment by konnov** [comment]
> Do we need to track `package-lock.json` via git? I thought this was autogen.

It is autogen but it's a good practice to add, in order to make builds between your machine and my machine reproducible

**Comment by Kukovec** [comment]
> > Do we need to track `package-lock.json` via git? I thought this was autogen.
> 
> It is autogen but it's a good practice to add, in order to make builds between your machine and my machine reproducible

I distinctly remember an issue with quint where this file would be modified by something and would always have to be stashed between git-commits

File: .github/workflows/main.yml (hunk 1 above)
Comment (Kukovec) [reply .github/workflows/main.yml:18]: removing the highlighted lines

**Comment by konnov** [comment]
> > > Do we need to track `package-lock.json` via git? I thought this was autogen.
> > 
> > 
> > It is autogen but it's a good practice to add, in order to make builds between your machine and my machine reproducible
> 
> I distinctly remember an issue with quint where this file would be modified by something and would always have to be stashed between git-commits

There are plenty of good answers on stack overflow btw: https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

**Pull request merged**